.github/workflows/cibuild.yml

```
@@ -3,6 +3,8 @@ name: CI
 on:
   push:
```

If you want to prevent people from kicking off CI on their own feature branches in their forks, you can make this stricter.
```suggestion
  push:
    branches:
    - master
    tags:
```

---

> The matrix could be trimmed. Is there a reason why you are testing against astropy 3.x when 4.0.x is the current LTS? I think the important astropy versions to test against are: 4.0.x (LTS), 4.2 (latest stable), and dev
> 
> So the matrix could be trimmed down to 3 jobs (any more you add would be a bonus but I am not sure how necessary):
> 
> 1. Python 3.7 + astropy 4.0.x + your oldest numpy (1.17?)
> 2. Python 3.8 + astropy 4.2 + latest stable numpy
> 3. Python 3.9 + astropy dev + latest/dev numpy (dev numpy might be overkill)
> 
> The coverage job: Can it be attached to one of the matrix jobs above (maybe stable astropy + stable numpy)? See `astropy` Actions + Tox setup for example.
> 
> Coverage reporting: I do see that it commented on the PR, so coverage upload works. `astropy` does have a `codecov.yml` that disables commenting but I don't think that is mutually exclusive with PR status posting. Maybe get this in `master` first and see if the status then magically appears in subsequent PRs.
> 
> Doc build job: I would say it is unnecessary if you opt to use RTD PR builder (which also allows doc preview). Up to you though.

Thanks @pllim 
I'm fine with dropping support for 3.x if that's `astropy` policy. Do we have agreement. As for `numpy`, as I've mentioned before, for the `numpy` used in this project I don't think we should bother with different versions. That's dealt with upstream (i.e. `astropy`)

I'm going to look into attaching coverage to one of the existing jobs in the matrix

How does the RTD builder app work? I haven't used it before.

---

Any reason not to do CI on feature branches? I use them in my fork and that's why I prefer to see the CI results after a push in a feature branch.

---

> How does the RTD builder app work?

Go to https://readthedocs.org/projects/pyvo/ -> Admin -> Advanced Settings (top left) -> check the "Build pull requests for this project" -> click "Save" (bottom)

Any push to PR after that would trigger a new job from RTD. It is controlled by your https://github.com/astropy/pyvo/blob/master/.readthedocs.yml but you do need to upgrade that YAML to RTD v2 first (https://docs.readthedocs.io/en/stable/config-file/v2.html). You also have the option to let it fail if there is any warning or not; up to you.

> Any reason not to do CI on feature branches?

To save the Earth a little, I guess. But if you think those extra runs are necessary, then keep it as-is.

---

As for dropping matrix for `astropy` 3.x, that is up to you. I don't see any minversion of `astropy` defined in your `setup.cfg`. Usually you want to test against the minversion that you say you support.

---

I agree with @pllim's suggestion on trimming the matrix, and see no reason to do anything with `astropy` 3.x.  I would even be OK with us specifying `astropy>=4.1` in `setup.cfg` since that's not a big ask from our users and gives us consistent handling of string values in VOTables (no more byte strings).

For our convenience, I do like triggering CI on feature branches.  It is overkill, but I have found it to save me headaches and dev time by catching weird things sooner.  That helps me rationalize the extra runs along with the fact that the `pyvo` test suite doesn't take long at all to run.

I yield to @pllim's opinion on the other matters, and admit to not fully grasping the details of the macOS and Windows build specs, but am happy to see how they run.

---

I guess it's up to all of us to decide on the minversion of `astropy`. Easiest is of course to support just the current version but we should be considerate of users that deploy `pyvo` in pipelines, virtual or containerized environments for which upgrading is not a minor task. Are there any `conda` constraints that we need to consider? How about direct dependencies such as `astroquery`? @bsipocz what is `astroquery` policy on minversions? Otherwise, we could decide to support only minversion that newer then 1 (or 2 or X) years, for example.
In other projects that I'm involved with, users are very slow in upgrading for the reasons above (no sysadmin access, old VM images, etc). I'm still maintaining software that is supposed to run Python2.7 and that becomes trickier! I'm surprised but happy that `astropy` is dropping support for older Python version at this rate.

---

Is there a way you can ask PyVO stakeholders about this? No one should still be using Python 2 or 3.5, so I don't see why anyone needs `astropy<4`.

---

I think that `astroquery` is the main stakeholder so I'll wait for @bsipocz word.

---

Can ping @keflavich and @ceb8 too.

---

I don't think we rely on astropy < 4; is that the question?  haven't time to read back through whole convo now

---

Yes, that was the question. In this case, we'll drop support for 3.*